easybuild/easyblocks/o/openssl_wrapper.py

```diff
@@ -0,0 +1,286 @@
+        """Fetch sources if OpenSSL component is needed"""
+        if not all([self.ssl_syslib, self.ssl_sysheader]):
+            super(EB_OpenSSL_wrapper, self).fetch_step(*args, **kwargs)
+
+    def extract_step(self):
+        """Extract sources if OpenSSL component is needed"""
+        if not all([self.ssl_syslib, self.ssl_sysheader]):
+            super(EB_OpenSSL_wrapper, self).extract_step()
+
+    def install_step(self):
+        """Symlink target OpenSSL installation"""
+
+        if self.ssl_syslib and self.ssl_sysheader:
+            # Link OpenSSL libraries in system
+            lib64_dir = os.path.join(self.installdir, 'lib64')
+            lib64_engines_dir = os.path.join(lib64_dir, os.path.basename(self.ssl_sysengines))
+            mkdir(lib64_engines_dir, parents=True)
+
+            lib_files = [lib_so for lib_so in self.openssl_libs]
+            lib_files = [os.path.join(os.path.dirname(self.ssl_syslib), '%s' % ptrn) for ptrn in lib_files]
+
+            engine_lib_pattern = [os.path.join(self.ssl_sysengines, '*')]
+            lib_files.extend(expand_glob_paths(engine_lib_pattern))
+
+            # link existing known libraries
+            for libso in lib_files:
+                if 'engines' in libso:
+                    target_dir = lib64_engines_dir
+                else:
+                    target_dir = lib64_dir
+                symlink(libso, os.path.join(target_dir, os.path.basename(libso)))
+
+            # link unversioned libraries
+            for libso in self.openssl_libs:
+                versioned_lib = os.path.join(lib64_dir, libso)
+                unversioned_lib = os.path.join(lib64_dir, '%s.%s' % (libso.split('.')[0], get_shared_lib_ext()))
+                symlink(versioned_lib, unversioned_lib)
+
+            # Link OpenSSL headers in system
+            include_dir = os.path.join(self.installdir, 'include', self.name.lower())
+            mkdir(include_dir, parents=True)
+
+            include_pattern = [os.path.join(self.ssl_sysheader, '*')]
+            include_files = expand_glob_paths(include_pattern)
+            for header in include_files:
+                symlink(header, os.path.join(include_dir, os.path.basename(header)))
+
+            # Link OpenSSL binary in system
+            bin_dir = os.path.join(self.installdir, 'bin')
+            mkdir(bin_dir)
+            symlink(self.ssl_sysbin, os.path.join(bin_dir, self.name.lower()))
+
+        else:
+            # Install OpenSSL component
+            super(EB_OpenSSL_wrapper, self).install_step()
+
+    def sanity_check_step(self):
+        """Custom sanity check for OpenSSL wrapper."""
+        shlib_ext = get_shared_lib_ext()
+
+        ssl_files = [os.path.join('bin', self.name.lower())]
+        ssl_files.extend(os.path.join('lib', libso) for libso in self.openssl_libs)
+        ssl_files.extend(os.path.join('lib', '%s.%s' % (libso.split('.')[0], shlib_ext)) for libso in self.openssl_libs)
```

Comment:
This doesn't work for a from-source installation of OpenSSL 1.0 on RHEL8, because there's no `lib/libssl.so.10`:
```
$ ls -l /apps/RHEL8/zen2-ib/software/OpenSSL/1.0/lib/
total 16756
drwxrwxr-x 2 vsc40023 vsc40023    4096 May 17 17:45 engines
-rw-r--r-- 1 vsc40023 vsc40023 4577474 May 17 17:45 libcrypto.a
lrwxrwxrwx 1 vsc40023 vsc40023      18 May 17 17:45 libcrypto.so -> libcrypto.so.1.0.0
-r-xr-xr-x 1 vsc40023 vsc40023 2707224 May 17 17:45 libcrypto.so.1.0.0
-rw-r--r-- 1 vsc40023 vsc40023  763794 May 17 17:45 libssl.a
lrwxrwxrwx 1 vsc40023 vsc40023      15 May 17 17:45 libssl.so -> libssl.so.1.0.0
-r-xr-xr-x 1 vsc40023 vsc40023  513616 May 17 17:45 libssl.so.1.0.0
drwxrwxr-x 2 vsc40023 vsc40023    4096 May 17 17:45 pkgconfig
```

I think we should just check for the `.so` (since that's a symlink to the actual library). If it's a broken symlink, the sanity check will fail.

Reply:
fixed in 2897480

Reply:
Then in this case the easyblock is not wrapping that installation of v1.0. We should add `libssl.so.1.0.0` as a possible file name

Reply:
fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/2429/commits/5701d618c3b6e97dbbf6635ede2cd15c43c33324 : now this easyblock can check multiple libraries to be wrapped per system/version and it also can properly sanity check the built component